components/kserve/kserve_config_handler.go

```diff
@@ -0,0 +1,148 @@
+package kserve
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+)
+
+const (
+	KserveConfigMapName string = "inferenceservice-config"
+)
+
+func (k *Kserve) setupKserveConfigAndDependencies(ctx context.Context, cli client.Client, dscispec *dsciv1.DSCInitializationSpec) error {
+	// as long as Kserve.Serving is not 'Removed', we will setup the dependencies
+
+	if k.Serving.ManagementState != operatorv1.Removed {
+		// check on dependent operators if all installed in cluster
+		dependOpsErrors := checkDependentOperators(cli).ErrorOrNil()
+		if dependOpsErrors != nil {
+			return dependOpsErrors
+		}
+
+		if err := k.configureServerless(dscispec); err != nil {
+			return err
+		}
+
+		if err := k.setDefaultDeploymentMode(ctx, cli, dscispec, k.DefaultDeploymentMode); err != nil {
+			return err
+		}
+	}
+
+	// since serverless is specifically set to 'Removed'
+	// we do not need to check if dependent operators are installed and configure them
+	// in this case the default deployment mode is forced to be RawDeployment
+	if k.DefaultDeploymentMode == Serverless {
```

Comment:
If Serverless is set to `Removed` but the default deploymentmode is `Serverless`, I'm not sure if it's better to just throw an error saying that this combination is incompatible or print a warning but go ahead and default to RawDeployment. If we default to RawDeployment in this case, is there a way to set `defaultDeploymentMode` in the DSC to `RawDeployment` ourselves? otherwise internally the deploymentmode will be `RawDeployment` but the DSC will still say `Serverless`, which might be misleading. 
@zdtsw @bartoszmajsak @israel-hdez

Reply:
IMHO, if the intent is explicit in the DSC, the right approach is to terminate with error.

Reply:
I would agree with that approach as well. 
@bartoszmajsak @zdtsw if there is consensus I can implement this

Reply:
does this make it easier to read:
```
switch k.Serving.ManagementState
case operatorv1.Managed, operatorv1.Unmanaged:
	// check on dependent operators if all installed in cluster
	dependOpsErrors := checkDependentOperators(cli).ErrorOrNil()
	if dependOpsErrors != nil {
		return dependOpsErrors
	}

	if err := k.configureServerless(dscispec); err != nil {
		return err
	}
case operatorv1.Removed:
	if k.DefaultDeploymentMode == Serverless {
		return fmt.Errorf("Misconfig: enable Serving in Kserve to use defaultDeployment mode with Serverless")
	} esle {
		fmt.Println("Serverless is specified as 'Removed', Kserve will default to RawDeployment")
	}


if err := k.setDefaultDeploymentMode(ctx, cli, dscispec, k.DefaultDeploymentMode); err != nil {
	return err
}
```

Reply:
@VedantMahabaleshwarkar @zdtsw Quick question: why are we checking on dependent operators when serving state is `Unmanaged` ?

Reply:
@VaishnaviHire cleaned up logic to only check dependent operators if serving is `Managed`